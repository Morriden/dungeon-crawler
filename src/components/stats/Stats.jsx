import React from 'react';
import PropTypes from 'prop-types';

const Stats = ({ health, attack, energy, experience }) => (
  <dl>
    <dt>Health</dt>
    <dd>{health}</dd>

    <dt>Attack</dt>
    <dd>{attack}</dd>

    <dt>Energy</dt>
    <dd>{energy}</dd>

    <dt>Experience</dt>
    <dd>{experience}</dd>
  </dl>
);

Stats.propTypes = {
  health: PropTypes.number.isRequired,
  attack: PropTypes.number.isRequired,
  energy: PropTypes.number.isRequired,
  experience: PropTypes.number.isRequired
};

export default Stats;
