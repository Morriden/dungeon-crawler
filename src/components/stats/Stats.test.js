import React from 'react';
import { shallow } from 'enzyme';
import Stats from './Stats';

describe('Stats component', () => {
  it('should render stat block', () => {
    const wrapper = shallow(<Stats health={10} attack={1} energy={10} experience={0} />);
    expect(wrapper).toMatchSnapshot();
  });
});
