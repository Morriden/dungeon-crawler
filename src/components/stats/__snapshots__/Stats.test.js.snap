// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats component should render stat block 1`] = `
<dl>
  <dt>
    Health
  </dt>
  <dd>
    10
  </dd>
  <dt>
    Attack
  </dt>
  <dd>
    1
  </dd>
  <dt>
    Energy
  </dt>
  <dd>
    10
  </dd>
  <dt>
    Experience
  </dt>
  <dd>
    0
  </dd>
</dl>
`;
